pkg/ccl/backupccl/restore_planning.go

```diff
@@ -718,12 +720,19 @@ func allocateDescriptorRewrites(
 				}
 
 				// See if there is an existing type with the same name.
+				getParentSchemaID := func(typ *typedesc.Mutable) (parentSchemaID descpb.ID) {
+					// If we find UDS with same name defined in the restoring DB, use its ID instead.
+					if parentSchemaID, ok = userDefinedSchemaIDMap[typ.GetParentSchemaID()]; !ok {
```

Comment:
Just wondering, instead of creating a new `userDefinedSchemaIDMap` can we just use `descriptorRewrites`?

Can we do something like:

```
parentSchemaID = typ.GetParentSchemaID()
if rewrite, ok := descriptorRewrites[typ.ID]; ok {
     // Check if an ID rewrite has been specified.
     if rewrite.ID != 0 {
         parentSchemaID = rewrite.ID
     }
}

Reply:
Good call!